lib/logger/lib/logger.ex

```diff
@@ -812,20 +813,22 @@ defmodule Logger do
     end
   end
 
+  defguardp is_msg(msg) when is_binary(msg) or is_list(msg) or is_map(msg)
+
   @doc false
   def __do_log__(level, fun, metadata) when is_function(fun, 0) and is_map(metadata) do
     case fun.() do
       {msg, meta} ->
-        :logger.macro_log(%{}, level, msg, Enum.into(meta, add_elixir_domain(metadata)))
+        __do_log__(level, msg, Enum.into(meta, add_elixir_domain(metadata)))
```

Comment:
Aren't we adding the elixir domain twice? Maybe we shouldn't call `add_elixir_domain` since we already do it in `do_log`.

Reply:
I will check that. And fix it if needed.

Reply:
Yeah, it was added twice. Fixed. I will also add test for that.